### DISCUSSION TITLE
Info About Merge sketch

### DISCUSSION THREAD

**Discussion by MLJ-ld** [sstaub/eOS#9] [Q&A]
Hi I would like to mix the sketch fader wing and cue box, I would use arduino mega, could it work?

**Comment by sstaub** [comment]
Should work with a Mega.

**Reply by MLJ-ld** [reply]
If I insert library LCD03 I could use i2c to connect the display and connecting only two pins, is it enough to eliminate too many constants or do you need to make other changes to the sketch?

**Comment by sstaub** [comment]
You must look only for conflicts with GPIO you use.

**Reply by MLJ-ld** [reply]
![image](https://user-images.githubusercontent.com/82513403/158429267-ed3ebaac-3944-4f2f-84b9-ad9959035e18.jpeg)
Hi,
I2c work good!
Now I work to merge your sketch

**Reply by MLJ-ld** [reply]
To best performance ( if necessary) could the code also work on Arduino due?  of course I would supply the 5 volts for display externally

**Comment by sstaub** [comment]
I think an Arduino Due will also work, but it shouldn't necessary. For more powerful things you can also use a Teensy 4.x. But both will not work only with Nomad.

**Reply by MLJ-ld** [reply]
I have seen the features of Teensy 4.1 !!  wow !!  I think I will use it to make the program wing in the next project with your sketches, as soon as the execution wing is finished.  thanks for the input !!

**Comment by sstaub** [comment]
Maybe that there are some problems with the OSC library with Teensy 4.1, If you get the 5V compatible Teensy 3.5 most things should work. I'm writing a complete new library but it will take some much more time to implement all goals.

**Reply by MLJ-ld** [reply]
For left side I use your sketch cue box + fader wing..![image](https://user-images.githubusercontent.com/82513403/159543951-c4c1f8f0-0ded-4e7e-872b-de3c46b663b4.jpeg)

**Comment by MLJ-ld** [comment]
HI,
I noticed that even if I send STOP_BACK from your sketch, the last word belongs to eos, because having set OFF on the STOP_BACK button (in fader config) the console performs an OFF and not STOP_BACK; and a correct compartment? shouldn't the console execute the OSC command that I send it?

**Reply by sstaub** [reply]
That is right.  The command refers to the hardware buttons.

**Reply by MLJ-ld** [reply]
Hi,
At moment Teensy 3.5 is out of stock, I can buy Teensy 3.6 or 4.1 this two hardware work gods with you sketch? For 5 volt display I can use external 5 volt from ps.

**Comment by sstaub** [comment]
The 4.1 have also Ethernet on board but I don't know if it works with the OSC library.

**Reply by MLJ-ld** [reply]
Ok, 3.6 can work? And in your skatch can I send all eos buttons? Or I must modify library?
Tks

**Comment by sstaub** [comment]
3.6 should work

**Comment by sstaub** [comment]
Teensy 4.1 works also, in setup() you need to comment out `//while (!SerialUSB);` because of blocking.
You can use the USB port for powering a display (Pin 5V on Teens 4.1).

**Comment by MLJ-ld** [comment]
This is good news, many thanks! I intend to create a program wing with your sketch by putting all the eos apex buttons and the two wheels with the change of parameters as he has already prepared in the sketch, all to be placed side by side with the EXE wing that I have done in these days. Thanks for the support
